core/src/main/java/io/questdb/cairo/wal/ApplyWal2TableJob.java

```diff
@@ -544,6 +551,31 @@ private int processWalCommit(
                     walTelemetryFacade.store(WAL_TXN_DATA_APPLIED, writer.getTableToken(), walId, s, walRowCount, commitPhRowCount, latency);
                     lastCommittedRows += walRowCount;
                 }
+
+                if (walTxnType == MAT_VIEW_DATA) {
+                    try (WalEventReader eventReader = walEventReader) {
```

Comment:
Few problems:
- The same code block duplicated in lines 609-630
- This branch processes multiple transactions, `(int) (writer.getAppliedSeqTxn() - seqTxn + 1)` to be precise but this only reads additional details from the first transactions. Please read the last applied transaction of `writer.getAppliedSeqTxn()` from WAL-E and update the status file with the latest applied.

Please refactor and read the last applied transaction values.

Reply:
Another problem is that at this point when transactions are marked as applied, there can be a race condition, and `WalPurgeJob` can delete the WAL-e files before we read them here.

The solution can be to pre-read `LastRefreshBaseTableTxn`, `LastRefreshTimestamp` into WalTxnDetails like other transaction parameters.

Reply:
fixed in: https://github.com/questdb/questdb/pull/5519/commits/6a53899131294e417b0980622cdc75de2dfb9e0b